src/utils.ts

```diff
@@ -17,6 +17,169 @@ const isObject = (value: unknown) =>
   !(value instanceof Error) &&
   !(value instanceof Date);
 
+// toCidr takes an IP and prefix length and returns a normalized CIDR.
+export function toCidr(ip: string, prefixLen: number) {
+  const isV6 = ip.includes(':');
+  const isV4Mapped = isV6 && ip.includes('.');
+  const cleanIp = isV4Mapped
+    ? ip.replace(/(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/, (_, v4) =>
+        binaryToV6Part(ipToBinary(v4, false))
+      )
+    : ip;
+
+  const binaryIp = ipToBinary(cleanIp, isV6);
+  const binaryMasked = maskPrefix(binaryIp, prefixLen);
+  const masked = isV6 ? binaryToV6(binaryMasked) : binaryToV4(binaryMasked);
```

Comment:
Since `binaryToV6` throws, we may want a try/catch block

Reply:
It would be a programming bug. I don't think we would want to catch it.